### PR TITLE
Merge SqlDelegatedTransaction

### PR DESCRIPTION
**Pull request by edwardneal**
Contributes to #1261.

This PR merges the SqlDelegatedTransaction class. It's slightly more complex, because it introduces a behaviour change in the .NET Framework class. I think the change is correct though.

Line 262 of the .NET Framework class was formerly:
```csharp
if(Transaction?.TransactionInformation?.Status == System.Transactions.TransactionStatus.Aborted)
```
It's now on line 257:
```csharp
if (Transaction?.TransactionInformation?.Status != TransactionStatus.Aborted)
```

This changes the .NET Framework class to align with the .NET Core codebase. Historically, both conditions were identical between codebases; PR #1216 changed this to fix a bug in .NET Core, in the process swapping the condition from `!=` to `==` in .NET Framework. I'm pretty sure this was just a typo, and the merge will correct that.

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlDelegatedTransaction.cs [hunk 1]
```diff
@@ -580,11 +579,15 @@ private void ValidateActiveOnConnection(SqlInternalConnection connection)
 
         // Get the server-side Global Transaction Id from the PromotedDTCToken
         // Skip first 4 bytes since they contain the version
+#if NET
+        private Guid GetGlobalTxnIdentifierFromToken() => new Guid(new ReadOnlySpan<byte>(_connection.PromotedDTCToken, _globalTransactionsTokenVersionSizeInBytes, 16));
```
Comment (benrr101) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlDelegatedTransaction.cs:583]: nit: Put the body on a new line?

**Review by benrr101 — CHANGES_REQUESTED** [review]
In general it looks good - let's get the merge conflicts resolved, take a look at the comments, make sure the build succeeds, and we'll be good to go!

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlDelegatedTransaction.cs [hunk 2]
```diff
@@ -87,12 +80,13 @@ public void Initialize()
             // transaction.
             SqlInternalConnection connection = _connection;
             SqlConnection usersConnection = connection.Connection;
-            SqlClientEventSource.Log.TryTraceEvent("<sc.SqlDelegatedTransaction.Initialize|RES|CPOOL> {0}, Connection {1}, delegating transaction.", ObjectID, connection.ObjectID);
+            SqlClientEventSource.Log.TryTraceEvent("SqlDelegatedTransaction.Initialize | RES | CPOOL | Object Id {0}, Client Connection Id {1}, delegating transaction.", ObjectID, usersConnection?.ClientConnectionId);
```
[the remark this answers is not in the document]
Comment (edwardneal) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlDelegatedTransaction.cs:83]: Thanks for this, I'm not sure on the long-term approach for tracing either; the merged codebase isn't consistent. SqlDataAdapter ([line 170](https://github.com/dotnet/SqlClient/blob/main/src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlDataAdapter.cs#L170)) uses the .NET Core trace style, while DbConnectionInternal ([line 249](https://github.com/dotnet/SqlClient/blob/main/src/Microsoft.Data.SqlClient/src/Microsoft/Data/ProviderBase/DbConnectionInternal.cs#L249)) uses the .NET Framework one. The SqlFileStream approach is best here, I think - I'll use that here and going forward.

**Comment by edwardneal** [comment]
Thanks @benrr101 - branch has been brought up to date, the trace logs left as-is and the nit around GetGlobalTxnIdentifierFromToken addressed!

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlDelegatedTransaction.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (David-Engel) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlDelegatedTransaction.cs]: I believe the `<` characters were leftover from the previous BID trace format and can be removed, when convenient, along with making the spacing consistent around `|`. They currently just make trace messages look ugly in netfx, IMO.

**Review by David-Engel — APPROVED** [review]

**Review by cheenamalhotra — APPROVED** [review]

**Review by benrr101 — APPROVED** [review]
Not sure how I missed updating this one, but we're good!

**Pull request merged**